src/domain_bridge/topic_bridge.hpp

```diff
@@ -45,27 +45,37 @@ struct TopicBridge
 
   /// The subscription half of the bridge
   std::shared_ptr<GenericSubscription> subscription;
-};
 
-/// Compare two TopicBridge objects
-struct TopicBridgeCompare
-{
-  bool operator()(const TopicBridge & lhs, const TopicBridge & rhs) const
+  /// Less-than operator.
+  /**
+   * Sort by 'from_domain_id',
+   *   then by 'to_domain_id',
+   *   then by 'topic_name',
+   *   then by 'type_name'
+   */
+  bool operator<(const TopicBridge & other) const
   {
-    // Sort by 'from_domain_id',
-    //   then by 'to_domain_id',
-    //   then by 'topic_name',
-    //   then by 'type_name'
-    if (lhs.from_domain_id < rhs.from_domain_id) {
+    if (from_domain_id < other.from_domain_id) {
       return true;
     }
-    if (lhs.to_domain_id < rhs.to_domain_id) {
+    if (from_domain_id != other.from_domain_id) {
+      return false;
+    }
+    if (to_domain_id < other.to_domain_id) {
       return true;
     }
-    if (lhs.topic_name.compare(rhs.topic_name) < 0) {
+    if (to_domain_id != other.to_domain_id) {
```

Comment:
For clarity, I would've used ">" here; it maintains the consistency with the less than instead of switching gears to !=.

Reply:
Sounds good to me: 37f1288